Add default values to usage

walle opened the pull request:
Check if the value isn't it's zero value and if not add a default value
to the usage text.

Updated implementation of default values in usage from discussion in #18

kentquirk commented:
It's up to @alexflint to decide if the format is right. Personally, I'd have put it in parentheses. But whatever. I like this change.

walle commented:
Good input @kentquirk I'm going to add parentheses around the default value, looks nicer and you see that it's a special value.

brettlangdon commented:
I think the general convention is to use brackets not parenthesis, see [docopt](http://docopt.org/)

But again, this is probably for @alexflint to decide.

walle commented:
Good point @brettlangdon, I've updated to brackets.

alexflint commented on usage.go:
```diff
@@ -95,6 +95,13 @@ func (p *Parser) WriteHelp(w io.Writer) {
+			v := spec.dest
+			z := reflect.Zero(v.Type())
+			if v.Interface() != z.Interface() {
+				fmt.Fprintf(w, " [default: %s]", v)
```
For non-strings, I believe this will end up printing something like `%!s(reflect.Value=1)`. Can you use `%q` instead?

walle replied:
Changed this to use %v instead. %q did not work for int (I added an int value to the usage test to verify that it works for non string values) it gave '*'

alexflint commented:
This looks good. Sorry for delay in responding (I'm actually getting married this weekend, so have been a bit busy!) One comment then happy to merge.

walle commented:
I have updated the code (and the test) to use %v instead of %s (see my line comment). 

Congratulations on getting married!

alexflint commented:
Thanks @walle. After conflicts are resolved I believe this will be good to go.

walle commented:
I've rebased and fixed the merge conflicts. I think I got it right, but would like another eye on it. The tests pass though.

Since the new code shows built ins, that don't read from spec.dest but spec.isBool, an extra check is needed to not try to read from the zero value of spec.dest. https://github.com/alexflint/go-arg/pull/19/files#diff-3472cb4e0a973362a7dbc843481a069eR107

alexflint commented:
Thanks @walle!
